pkg/bsp-imx/patches/uboot-patches-lf_v2022.04/0003-arch-arm-dts-imx8mq-evk-add-fdt-kernel-node-to-binma.patch

```diff
@@ -0,0 +1,46 @@
+From fe9962e501632b58e04c896a523140fd8acb1312 Mon Sep 17 00:00:00 2001
+From: Insei <goodmobiledevices@gmail.com>
+Date: Thu, 8 Dec 2022 05:38:41 +0300
+Subject: [PATCH] arch: arm: dts: imx8mq-evk: add fdt-kernel node to binman
+ configuration
+
+Signed-off-by: Insei <goodmobiledevices@gmail.com>
+---
+ arch/arm/dts/imx8mq-u-boot.dtsi | 14 +++++++++++++-
+ 1 file changed, 13 insertions(+), 1 deletion(-)
+
+diff --git a/arch/arm/dts/imx8mq-u-boot.dtsi b/arch/arm/dts/imx8mq-u-boot.dtsi
+index 1dc060ce0c..408b0cc8d8 100644
+--- a/arch/arm/dts/imx8mq-u-boot.dtsi
++++ b/arch/arm/dts/imx8mq-u-boot.dtsi
+@@ -103,6 +103,18 @@
+ 						type = "blob-ext";
+ 					};
+ 				};
++
++				fdt-kernel {
++					compression = "none";
++					description = "NAME";
++					type = "flat_dt";
++					load = <0x43000000>;
++
++					kernel-fdt-blob {
++						filename = "arch/arm/dts/imx8mq-evk.dtb";
```

Comment:
Your are fixing the device tree for one particular board in the generic platform device tree. This can be a problem if we want to build two u-boots of the same platform for two different boards. For instance, in my case I want to build the same u-boot code base for Pollux and for another device (both with imx8mp SoC). Thus, I cannot fix the kernel device tree in imx8mp-u-boot.dtsi. My approach was to let the kernel device tree outside the image (in EFI partition), so it can be loaded by bootcmd command (I've enabled GPT support in order to be able to do that). Have a look here: https://github.com/lf-edge/eve/commit/dd47c39c15b428811060313fb15ad4d25a6a04c6#diff-7715c1c76c3810b834e7d2308ab9dd7eb3d51db13db8782514a8731c786195f8

Please, take a look and think about it. I want to see this PR merged soon, so if you think is too much effort to change here, that's fine, let's let as it is for now...

Reply:
@rene This is a good observation. But I think I'll just move this node to imx8mq-evk-u-boot.dtsi, because I think this option is easier to implement. In the option of storing this blob on the EFI partition, it adds complexity for the future implementation of the update process - we will have to update both the blob on the disk and the dtb in the partition, I think it would be more convenient to update one thing. That's why I put everything together in one blob.

Reply:
I see. That's fine. Indeed, I think move to the board's dts (imx8mq-evk-u-boot.dts) would be better....